### ISSUE TITLE
ArgumentError: wrong number of arguments

### ISSUE THREAD

**Issue by pmietlicki** [dhanasingh/redmine_wktime#106]
Hi,
Under redmine 3.4.2 when I try to install the plugin :
```
root@0d96f4453818:/usr/src/redmine# rake redmine:plugins:migrate NAME=redmine_wk
time RAILS_ENV=production
rake aborted!
ArgumentError: wrong number of arguments (given 0, expected 1+)
/usr/src/redmine/plugins/redmine_wktime/init.rb:130:in `include'
/usr/src/redmine/plugins/redmine_wktime/init.rb:130:in `included'
/usr/src/redmine/plugins/redmine_wktime/init.rb:238:in `include'
/usr/src/redmine/plugins/redmine_wktime/init.rb:238:in `<top (required)>'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:27
4:in `require'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:27
4:in `block in require'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:24
0:in `load_dependency'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:27
4:in `require'
/usr/src/redmine/lib/redmine/plugin.rb:155:in `block in load'
/usr/src/redmine/lib/redmine/plugin.rb:146:in `each'
/usr/src/redmine/lib/redmine/plugin.rb:146:in `load'
```
Do you have any idea about this bug ?
I don't know much in ruby on rails...
Thanks a lot.

Best regards,
Pascal

**Comment by dhanasingh** [comment]
What version of ERPmine are you installing ?

**Comment by pmietlicki** [comment]
The 2.9.1, the last one, the one that is on the official redmine plugin page. I've also tried downloading a zip from github directly, but I have the same error.

Thanks.

**Comment by dhanasingh** [comment]
Ok ERPmine is compatible with Redmine 3.4, we have tested with 3.4.1, not with 3.4.2,  is it possible to check with Redmine 3.4.1 ?

Can you give us versions of OS, Ruby, Rails, database etc ?

**Comment by pmietlicki** [comment]
Hi,
I'm on 3.4.2 and I'd like to stay on this version.
I'm using docker with the official [redmine:passenger](https://github.com/docker-library/redmine/blob/ef5e3f30a26cb30d91d940019c0e8719e3007eb3/3.4/Dockerfile) repository, the database is galera (mariadb cluster) with ruby 2.4-slim

Thank you for your help.

**Comment by DanielBoettner** [comment]
Same problem here, on our testserver.

here is a full trace of the migrate task

``` 
ArgumentError: wrong number of arguments (given 0, expected 1+)
/usr/src/redmine/plugins/redmine_wktime/init.rb:130:in `include'
/usr/src/redmine/plugins/redmine_wktime/init.rb:130:in `included'
/usr/src/redmine/plugins/redmine_wktime/init.rb:238:in `include'
/usr/src/redmine/plugins/redmine_wktime/init.rb:238:in `<top (required)>'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:274:in `require'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:274:in `block in require'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:240:in `load_dependency'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:274:in `require'
/usr/src/redmine/lib/redmine/plugin.rb:155:in `block in load'
/usr/src/redmine/lib/redmine/plugin.rb:146:in `each'
/usr/src/redmine/lib/redmine/plugin.rb:146:in `load'
/usr/src/redmine/config/initializers/30-redmine.rb:21:in `<top (required)>'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `load'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `block in load'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:240:in `load_dependency'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `load'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/engine.rb:652:in `block in load_config_initializer'
/usr/local/bundle/gems/activesupport-4.2.8/lib/active_support/notifications.rb:166:in `instrument'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/engine.rb:651:in `load_config_initializer'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/engine.rb:616:in `block (2 levels) in <class:Engine>'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/engine.rb:615:in `each'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/engine.rb:615:in `block in <class:Engine>'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/initializable.rb:30:in `instance_exec'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/initializable.rb:30:in `run'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/initializable.rb:55:in `block in run_initializers'
/usr/local/lib/ruby/2.4.0/tsort.rb:228:in `block in tsort_each'
/usr/local/lib/ruby/2.4.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
/usr/local/lib/ruby/2.4.0/tsort.rb:422:in `block (2 levels) in each_strongly_connected_component_from'
/usr/local/lib/ruby/2.4.0/tsort.rb:431:in `each_strongly_connected_component_from'
/usr/local/lib/ruby/2.4.0/tsort.rb:421:in `block in each_strongly_connected_component_from'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/initializable.rb:44:in `each'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/initializable.rb:44:in `tsort_each_child'
/usr/local/lib/ruby/2.4.0/tsort.rb:415:in `call'
/usr/local/lib/ruby/2.4.0/tsort.rb:415:in `each_strongly_connected_component_from'
/usr/local/lib/ruby/2.4.0/tsort.rb:349:in `block in each_strongly_connected_component'
/usr/local/lib/ruby/2.4.0/tsort.rb:347:in `each'
/usr/local/lib/ruby/2.4.0/tsort.rb:347:in `call'
/usr/local/lib/ruby/2.4.0/tsort.rb:347:in `each_strongly_connected_component'
/usr/local/lib/ruby/2.4.0/tsort.rb:226:in `tsort_each'
/usr/local/lib/ruby/2.4.0/tsort.rb:205:in `tsort_each'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/initializable.rb:54:in `run_initializers'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/application.rb:352:in `initialize!'
/usr/src/redmine/config/environment.rb:14:in `<top (required)>'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/application.rb:328:in `require'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/application.rb:328:in `require_environment!'
/usr/local/bundle/gems/railties-4.2.8/lib/rails/application.rb:457:in `block in run_tasks_blocks'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/task.rb:250:in `block in execute'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/task.rb:250:in `each'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/task.rb:250:in `execute'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/task.rb:194:in `block in invoke_with_call_chain'
/usr/local/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/task.rb:187:in `invoke_with_call_chain'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/task.rb:216:in `block in invoke_prerequisites'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/task.rb:214:in `each'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/task.rb:214:in `invoke_prerequisites'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/task.rb:193:in `block in invoke_with_call_chain'
/usr/local/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/task.rb:187:in `invoke_with_call_chain'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/task.rb:180:in `invoke'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/application.rb:152:in `invoke_task'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/application.rb:108:in `block (2 levels) in top_level'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/application.rb:108:in `each'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/application.rb:108:in `block in top_level'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/application.rb:117:in `run_with_threads'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/application.rb:102:in `top_level'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/application.rb:80:in `block in run'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/application.rb:178:in `standard_exception_handling'
/usr/local/bundle/gems/rake-12.0.0/lib/rake/application.rb:77:in `run'
/usr/local/bundle/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/usr/local/bundle/bin/rake:17:in `load'
/usr/local/bundle/bin/rake:17:in `<main>'
Tasks: TOP => redmine:plugins:migrate => environment
```

**Comment by dhanasingh** [comment]
can you try commenting these 2 lines from init.rb and see if it works fine
line 130		base.send(:include)
line 179		base.send(:include)

**Comment by DanielBoettner** [comment]
will do ASAP

**Comment by DanielBoettner** [comment]
Doesn't work

```
rake aborted!
NoMethodError: undefined method `before_filter' for WkreportController:Class
Did you mean?  before_action
/var/www/redmine/plugins/redmine_wktime/app/controllers/wkreport_controller.rb:28:in `<class:WkreportController>'
/var/www/redmine/plugins/redmine_wktime/app/controllers/wkreport_controller.rb:18:in `<top (required)>'
/var/www/redmine/plugins/redmine_wktime/init.rb:349:in `<top (required)>'
/var/www/redmine/lib/redmine/plugin.rb:155:in `block in load'
/var/www/redmine/lib/redmine/plugin.rb:146:in `each'
/var/www/redmine/lib/redmine/plugin.rb:146:in `load'
/var/www/redmine/config/initializers/30-redmine.rb:21:in `<top (required)>'
/var/www/redmine/config/environment.rb:14:in `<top (required)>'
Tasks: TOP => redmine:plugins:migrate => environment
(See full trace by running task with --trace)
```

**Comment by dhanasingh** [comment]
great, we will put this fix in the next release of ERPmine (3.0), until then please use this temporary fix
